metricbeat/module/system/service/service_test.go

```diff
@@ -126,3 +126,28 @@ func TestUnitStateNoFilter(t *testing.T) {
 	shouldReturnResults := matchUnitState([]string{}, exampleUnits)
 	assert.Len(t, shouldReturnResults, 3)
 }
+
+// func TestFetch(t *testing.T) {
+// 	f := mbtest.NewReportingMetricSetV2Error(t, getConfig())
+// 	events, errs := mbtest.ReportingFetchV2Error(f)
+
+// 	assert.Empty(t, errs)
+// 	if !assert.NotEmpty(t, events) {
+// 		t.FailNow()
+// 	}
+// 	t.Logf("Events: %d", len(events))
+
+// 	for _, unit := range events {
+// 		t.Logf("Unit: %s State: %s, SubState: %#v", unit.MetricSetFields["name"], unit.MetricSetFields["state"], unit.MetricSetFields["sub_state"])
+// 	}
+// }
+
+// func getConfig() map[string]interface{} {
+// 	return map[string]interface{}{
+// 		"module":                     "system",
+// 		"metricsets":                 []string{"service"},
+// 		"service.track_non_service":  false,
+// 		"service.track_not_found":    false,
+// 		"service.track_instantiated": true,
+// 	}
+// }
```

Comment:
Remove commented out code if not useful, though in this case maybe you can uncomment it and skip if systemd is not available.